src/util/cilfacade.ml

```diff
@@ -410,3 +380,89 @@ let countLoc fn =
   let res = Hashtbl.length locs in
   Hashtbl.clear locs;
   res
+
+
+
+module StmtH = Hashtbl.Make (CilType.Stmt)
```

Comment:
All these things work under the assumption that we do not add new globals to `current_file`, while the old implementations did not assume this. I think currently we do not rely on creating new globals, but we should document this here.

Reply:
They did though, all of them except `Cilfacade.getFun` constructed some kind of mapping once at the beginning, either during `MyCFG.createCFG`, using manually written laziness of `getdec` or `lazy`.

They all iterate over `GFun`s, so even if someone added those later, there would be bigger problems first: like no CFG for them, because `MyCFG.createCFG` works the same way.